debezium-connector-oracle/src/test/java/io/debezium/connector/oracle/OracleConnectorIT.java

```diff
@@ -947,6 +948,220 @@ public void shouldPropagateSourceTypeByDatatype() throws Exception {
+
+            // create table
+            connection.execute(ddl);
+            connection.execute("GRANT SELECT ON debezium.interval to " + TestHelper.getConnectorUserName());
+            connection.execute("ALTER TABLE debezium.interval ADD SUPPLEMENTAL LOG DATA (ALL) COLUMNS");
+
+            // Insert a snapshot record
+            connection.execute("INSERT INTO debezium.interval (id, intYM, intYM2, intDS, intDS2) "
+                    + "values (1, INTERVAL '2' YEAR, INTERVAL '555-4' YEAR(3) TO MONTH, "
+                    + "INTERVAL '3' DAY, INTERVAL '111 10:09:08.555444333' DAY(3) TO SECOND(9))");
+            connection.execute("INSERT INTO debezium.interval (id, intYM, intYM2, intDS, intDS2) "
+                    + "values (2, INTERVAL '0' YEAR, INTERVAL '0' MONTH, "
+                    + "INTERVAL '0' DAY, INTERVAL '0' SECOND)");
+            connection.commit();
+
+            final Configuration config = TestHelper.defaultConfig()
+                    .with(OracleConnectorConfig.TABLE_INCLUDE_LIST, "DEBEZIUM.INTERVAL")
+                    .with(OracleConnectorConfig.LOG_MINING_STRATEGY, "online_catalog")
+                    .with(OracleConnectorConfig.INTERVAL_HANDLING_MODE,
+                            OracleConnectorConfig.IntervalHandlingMode.STRING.getValue())
+                    .build();
+
+            // Perform a basic startup & initial snapshot of data
+            start(OracleConnector.class, config);
+            assertConnectorIsRunning();
+
+            waitForSnapshotToBeCompleted(TestHelper.CONNECTOR_NAME, TestHelper.SERVER_NAME);
+
+            // Verify record generated during snapshot
+            final List<SourceRecord> records = consumeRecordsByTopic(2).recordsForTopic("server1.DEBEZIUM.INTERVAL");
+            assertThat(records.size()).isEqualTo(2);
+            records.forEach(rec -> {
+                Struct after = ((Struct) rec.value()).getStruct(AFTER);
+                Struct key = (Struct) rec.key();
+                switch ((int) key.get("ID")) {
+                    case 1:
+                        assertThat(after.getString("INTYM")).isEqualTo("P2Y0M0DT0H0M0S");
+                        assertThat(after.getString("INTYM2")).isEqualTo("P555Y4M0DT0H0M0S");
+                        assertThat(after.getString("INTDS")).isEqualTo("P0Y0M3DT0H0M0S");
+                        assertThat(after.getString("INTDS2")).isEqualTo("P0Y0M111DT10H9M563.444333S");
+                        break;
+                    case 2:
+                        assertThat(after.getString("INTYM")).isEqualTo("P0Y0M0DT0H0M0S");
+                        assertThat(after.getString("INTYM2")).isEqualTo("P0Y0M0DT0H0M0S");
+                        assertThat(after.getString("INTDS")).isEqualTo("P0Y0M0DT0H0M0S");
+                        assertThat(after.getString("INTDS2")).isEqualTo("P0Y0M0DT0H0M0S");
+                        break;
+                    default:
+                        fail("unexpected id");
+                }
```

Comment:
I might be missing something, but why is it you don't do this:

```
SourceRecord record = records.get(0);
// assertions for record with PK 1

record = records.get(1);
// assertions for record with PK 2
```

Events will be ordered as per insertion order, so this should be a bit more straight forward?

Reply:
I was unsure about the ordering guarantee. But if that's the case this would make it simpler indeed.

Reply:
No worries @TomBillietKlarrio I've already adjusted this in a commit I'll push to this PR if that's okay.

Reply:
ah, I just did the same but you were a few minutes faster. Thanks for updating.